oauth_request.go

```diff
@@ -0,0 +1,53 @@
+package geddit
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+type oauthRequest struct {
+	accessToken string
+	url         string
+	useragent   string
+	values      *url.Values
+}
+
+func (r oauthRequest) getResponse() (*bytes.Buffer, error) {
+	// Determine the HTTP action.
+	var action, finalurl string
+	if r.values == nil {
+		action = "GET"
+		finalurl = r.url
+	} else {
+		action = "POST"
+		finalurl = r.url + "?" + r.values.Encode()
+	}
+
+	// Create a request and add the proper headers.
+	req, err := http.NewRequest(action, finalurl, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("User-Agent", r.useragent)
+	req.Header.Set("Authorization", "bearer "+r.accessToken)
+
+	// Handle the request
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
Thanks. I like `doRequest` as a function name better, especially since it will be accepting a request type as its argument. Other ideas include `getBody`, `getResponseBody`, but I don't think those names imply we will be performing a request in the process of getting the response body. 

Maybe `sendRequest` is the best name I can come up with for now. I think I still prefer `doRequest`, however. Something about the word "send" doesn't really imply that the caller will be getting something back.